src/components/Graph/MainGraph.js

```diff
@@ -83,87 +83,99 @@ class MainGraph extends Component {
-        const dateString = firstScenario.key.substring(0,10)
-        const dateThreshold = parseDate(dateString)
-
-        // firstSeverity need to be designated in case not all death rate LEVELS exist
-        const dates = dataset[firstScenario.key].dates.map( d => parseDate(d));
-        const series = dataset[firstScenario.key][firstSeverity][firstIndicator.key]
-            .slice(0, numDisplaySims);
-        const severityList = buildSeverities(scenarioMap, [], firstScenario.key);
-        const sevList = _.cloneDeep(severityList);
-        sevList[0].scenario = firstScenario.key;
-
-        // allSims used for R0 histogram
-        const allSims = dataset[firstScenario.key][firstSeverity][firstIndicator.key];
-
-        // set dateRange to a default based on equal padding around date of scenario run
-        const currIdx = dates.findIndex(date => formatDate(date) === formatDate(dateThreshold))
-        const datePadding = dates.length - currIdx
-        const startIdx = dates.length - 1 - (datePadding * 2)
-        
-        // have a multiple of ten pad each side of the dateRange - alternative way
-        // const numDates = dates.length
-        // const dateMargin =  Math.ceil(Math.ceil(numDates / 10) / 10) * 10
-        const dateRange = [dates[startIdx], dates[dates.length - 1]]
-
-        // initialize Threshold and slider ranges
-        const idxMin = timeDay.count(dates[0], dateRange[0]);
-        const idxMax = timeDay.count(dates[0], dateRange[1]);
-        const [indicatorThreshold, seriesMin, seriesMax] = getindicatorThreshold(
-            [firstScenario], [series], idxMin, idxMax);
-        const simsOver = flagSims(
-            series, indicatorThreshold, dates, dateThreshold)        
-        const newSelectedDates = Array.from(dates).slice(idxMin, idxMax);
-        const filteredSeries = filterByDate(series, idxMin, idxMax)
-
-        const confBoundsList = getConfBounds(
-            dataset, [firstScenario], severityList, firstIndicator, dates, idxMin, idxMax)
+        if (Object.keys(indicators).length === 0) {
+            console.log('MainGraph Error: indicators is empty.')
+        }
 
-        const actualList = getActuals(actuals, firstIndicator, [firstScenario]);
+        if (Object.keys(actuals).length === 0) {
+            console.log('MainGraph Error: actuals is empty.')
+        }
 
-        const r0full = getR0range(dataset, firstScenario, sevList[0], firstIndicator);
-        // seriesListForBrush used by handleBrush to initialize instead of R0 filtering 
-        // series is updated and set to state in scenario, sev, indicator, r0 change handlers
-        const seriesListForBrush = filterR0(
-            r0full, [firstScenario], sevList, firstIndicator, dataset, numDisplaySims);
+        if (Object.keys(dataset).length > 0) {
+            // SCENARIOS: various scenario variables used for a given geoid
+            const SCENARIOS = buildScenarios(dataset);  
```

Comment:
@BenjiLee is this a janky way to expose errors?

Reply:
Sort of. I think you need all of indicators, actuals and dataset. The way you have it now, the code will crash if  you are  missing indicators, actuals but have a dataset. 

You need to check them all to run the logic

Reply:
I think I have addressed this comment with the latest commit.